hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java

```diff
@@ -263,8 +290,9 @@ protected static void assertHasNoListenerEventually(String jobIdString, UUID reg
         assertTrueEventually(() -> {
             assertTrue(Arrays.stream(instances()).allMatch(hz ->
                     getEventService(hz).getRegistrations(JobEventService.SERVICE_NAME, jobIdString).isEmpty()));
-            assertFalse(((ClientListenerServiceImpl) getHazelcastClientInstanceImpl(client()).getListenerService())
-                    .getRegistrations().containsKey(registrationId));
+            assertTrue(registrationId == null
+                    || !((ClientListenerServiceImpl) getHazelcastClientInstanceImpl(client()).getListenerService())
```

Comment:
this may fix error in `@After` but the tests still will fail if the job ends before listener is registered (it will not get any event or too few or listener registration will throw). Is it possible in heavily loaded CI server.